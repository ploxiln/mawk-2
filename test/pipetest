#!/bin/sh

AWK=${AWK:-../mawk}

# pipes and out files and new printf and system

echo '3
2
1' |  $AWK 'BEGIN {
            input = "cat; exit 7"
	    output = "sort"

	    while((input | getline > 0)) {
	        print | output
	    }
	    print close(input), close(output)
}'

$AWK 'BEGIN {  
    file = "foo" rand()
    printf "%x %x %d\n", -1, 2^32, -2^32 > file
    print close(file)
    command = sprintf("cat %s; rm %s; exit 4", file, file)
    print system(command)
}'




