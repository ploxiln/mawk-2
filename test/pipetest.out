1
2
3
7 0
0
ffffffffffffffff 100000000 -4294967296
4
